src/cls/IPM/ResourceProcessor/WebApplication.cls

```diff
@@ -216,15 +216,23 @@ ClassMethod DoPropertiesExist(pPropList As %Library.List) As %Library.List [ Int
 
 Method ConstructRoleName(pDbResource As %String) As %String [ Internal, Private ]
 {
-    // Include the module name in the role name to avoid conflicts
+    // Include the module name in the role name to avoid conflicts, using hash because role name can't be longer than 64 characters
     // This role should only be managed by IPM itself
-    Return $ListToString($lb("IPM", "readonly", ..ResourceReference.Module.Name, pDbResource), ".")
+    Set hashBytes = $System.Encryption.MD5Hash(..ResourceReference.Module.Name) // 128 bits
```

Comment:
MD5 is not available when using FIPS mode so maybe not the best function to use? I recall a customer reporting this as being problematic in the past.
Is there a reason to not just create `%DB_<database>_READ` role? I think that's simpler and cleaner. Description can indicate it was created by IPM. Creating many duplicate roles doing the same thing for each module seems like overkill. 
The downside is the role can't be cleaned up but one additional (fairly useful) role is worth the simplicity and not creating excess config to manage IMO

@isc-shuliu @isc-tleavitt thoughts?

Reply:
Now that I think about it, if we want to automatically delete the role, the role should be listed as a standalone resource. So I agree we can simplify it and move the handling to %IPM.Utils.Module::%Evaluate() and create the role there.

@isc-tleavitt Does that make sense?

Reply:
The reason for a package-specific role is so that a package can clean up after itself. But fair enough to create a reusable READ role by convention.